components/test-suite/src/loci/tests/testng/Configuration.java

```diff
@@ -260,32 +262,33 @@ public Length getPhysicalSizeX() {
     String physicalSize = currentTable.get(PHYSICAL_SIZE_X);
     String sizeXUnits = currentTable.get(PHYSICAL_SIZE_X_UNIT);
     try {
-      return physicalSize == null ? null : FormatTools.getPhysicalSize(new Double(physicalSize), sizeXUnits);
-    }
-    catch (NumberFormatException e) {
-      return null;
+      return physicalSize == null ? null : UnitsLength.create(new Double(physicalSize), UnitsLength.fromString(sizeXUnits)); 
```

Comment:
The `FormatTools.getPhysicalSize()` also includes the knowledge of the default unit for the physical sizes (`UNITS.MICROMETER`). Here, if `PHYSICAL_SIZE_X_UNIT` is not specified in the configuration table e.g. for old configuration files, the getter will return `null` and cause the tests to fail - see https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-breaking-repository-subset/292/console.

To reduce the number of false positives, the unit length creation and`EnumerationException` should probably be handled separately. Eventually we might want to enforce the unit specification in the configuration files but at this is probably too much for the scope of this PR.